### ISSUE TITLE
Type PostgrestError is not assignable to class PostgrestError

### ISSUE THREAD

**Issue by ghost** [supabase/supabase-js#1287]
As we and probably most use postgrest-js through supabase-js, I also opened a bug here. The bug report for postgrest-js: https://github.com/supabase/postgrest-js/issues/561

# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

With the latest release v2.45.5 when importing PostgrestError from supabase-js, the PostgrestError class (https://github.com/supabase/postgrest-js/blob/master/src/PostgrestError.ts) is now imported instead of the PostgrestError type (https://github.com/supabase/postgrest-js/blob/master/src/types.ts) which are not compatible:

>  Type `PostgrestError` is not assignable to type `PostgrestError`.

This change was introduced with this pull request: https://github.com/supabase/postgrest-js/pull/555

## To reproduce

Write a generic function with a PostgrestError object as a parameter:

```
import { PostgrestError } from '@supabase/supabase-js'

export function handlePostgrestError(error: PostgrestError) {
  // do stuff
}
```

Then try to use this function with an error object returned by, e.g., a Supabase query:

```
import { createClient } from '@supabase/supabase-js'

const supabase = createClient<Database>(
  import.meta.env.SUPABASE_URL,
  import.meta.env.SUPABASE_KEY,
)

function loadData(someId: string) {
  const { data, error } = await supabaseAdmin
    .from('some_table')
    .select('some_column')
    .eq('id', someId)
    .single()

  if (error) {
    handlePostgrestError(error)
    return
  }

  ...
}
```

## Expected behavior

Type-check is successful

## System information
- Version of supabase-js: v2.45.5

**Comment by maximilian-hammerl** [comment]
My PR for postgrest-js https://github.com/supabase/postgrest-js/pull/562 was merged, and the changes have already been released in v1.16.3
Thanks @soedirgo for the quick response!

I opened a PR to update the postgrest-js dependency: https://github.com/supabase/supabase-js/pull/1289

**Comment by maximilian-hammerl** [comment]
Thanks for the quick response, problem is fixed with the newest release! :) Issue can be closed
